scripts/poweron/wlan.py

```diff
@@ -1,34 +1,55 @@
-#!/usr/bin/env python
+#!/usr/bin/env python3
 
 # Script which shows how to use the XenAPI to find a particular Host's management interface
-# and send it a wake-on-LAN packet. Used for the power-on fucntionality as well.
+# and send it a wake-on-LAN packet. Used for the power-on functionality as well.
 
-import subprocess, sys, socket, struct, time, syslog
-
-import XenAPI, inventory
+import socket
+import struct
+import subprocess
+import syslog
+import time
 
+import inventory
 import XenAPIPlugin
 
 
 def doexec(args, inputtext=None):
     """Execute a subprocess, then return its return code, stdout and stderr"""
-    proc = subprocess.Popen(args,stdin=subprocess.PIPE,stdout=subprocess.PIPE,stderr=subprocess.PIPE,close_fds=True)
-    (stdout,stderr) = proc.communicate(inputtext)
+    proc = subprocess.Popen(
+        args,
+        stdin=subprocess.PIPE,
+        stdout=subprocess.PIPE,
+        stderr=subprocess.PIPE,
+        universal_newlines=True,
+        close_fds=True,
+    )
+    (stdout, stderr) = proc.communicate(inputtext)
```

Comment:
These two variables contain bytes because the Popen didn't open them in text mode:

> If encoding or errors are specified, or text is true, the file objects stdin, stdout and stderr are opened in text mode with the specified encoding and errors, as described above in [Frequently Used Arguments](https://docs.python.org/3/library/subprocess.html#frequently-used-arguments). The universal_newlines argument is equivalent to text and is provided for backwards compatibility. By default, file objects are opened in binary mode.

https://docs.python.org/3/library/subprocess.html#subprocess.Popen

Please specify the encoding in Popen to fix this